backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/settings/interceptor/ModCntr_Module.java

```diff
@@ -107,19 +107,18 @@ private void validateProductInPS(@Nullable final ProductId productId, @NonNull f
 		}
 	}
 
-	@ModelChange(timings = { ModelValidator.TYPE_AFTER_CHANGE, ModelValidator.TYPE_AFTER_NEW })
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE })
```

Comment:
I'm afraid switching to BEFORE_XXX without adjusting the code below will cause problems

```java
	final boolean bothSalesOnRawProductAndProcessedProductSet = settings.getModuleConfigs()
				.stream()
				.filter(module -> module.isMatching(SalesOnRawProduct)
						|| module.isMatching(SalesOnProcessedProduct))
				.count() > 1;
```

Reply:
Didn't test at all yet, but this is the first on my list, because I'm not 100% sure how the diff is

Reply:
@cp-ps  @mrie04  shall it be `>0` instead?

Reply:
doesn't work on change then. reverted this. Problem in UI I could also force with this change